tests/test_helpers.py

```diff
@@ -975,3 +975,59 @@ def test_populate_with_cookies():
 )
 def test_parse_http_date(value, expected):
     assert parse_http_date(value) == expected
+
+
+@pytest.mark.parametrize(
+    ["netrc_contents", "hostname", "expected_username"],
+    [
+        (
+            "machine example.com login username password pass\n",
+            "example.com",
+            "username",
+        ),
+    ],
+    indirect=("netrc_contents",),
+)
+@pytest.mark.usefixtures("netrc_contents")
+def test_netrc_from_env(netrc_contents: str, hostname: str, expected_username: str):
+    """Test that reading netrc files from env works as expected"""
+    netrc_obj = helpers.netrc_from_env()
+    assert netrc_obj.authenticators(hostname)[0] == expected_username
+
+
+@pytest.mark.parametrize(
+    ["netrc_contents", "hostname", "expected_auth"],
+    [
+        (
+            "machine example.com login username password pass\n",
+            "example.com",
+            helpers.BasicAuth("username", "pass"),
+        ),
+        (
+            "machine example.com account username password pass\n",
+            "example.com",
+            helpers.BasicAuth("username", "pass"),
+        ),
+        (
+            "machine example.com password pass\n",
+            "example.com",
+            helpers.BasicAuth("", "pass"),
+        ),
+        ("", "example.com", None),
+    ],
+    indirect=("netrc_contents",),
+)
+@pytest.mark.usefixtures("netrc_contents")
+def test_basicauth_from_netrc(
+    netrc_contents: str,
+    hostname: str,
+    expected_auth: helpers.BasicAuth,
+):
+    """Test that netrc file contents are properly parsed into BasicAuth tuples"""
+    netrc_obj = helpers.netrc_from_env()
```

Comment:
Looks like this could be a module-local fixture.

Reply:
Given that it is only used twice, I'd *love* to keep it as is. My brain already hurts from the amount of indirection going on here with the parametrization, I'd love to let this one be.